toolchain/base/value_store.h

```diff
@@ -170,63 +140,63 @@ class ValueStore
   // ```
   // for (auto [id, value] : store.enumerate()) { ... }
   // ```
-  auto enumerate() const -> auto {
-    UnpoisonAll();
+  auto enumerate() const [[clang::lifetimebound]] -> auto {
```

Comment:
Just to note, if you're good with the `values()` approach, you could switch this to directly do a map on `llvm::seq` instead of the indirection through `values()`